src/main/java/com/yelp/nrtsearch/server/luceneserver/search/query/multifunction/FilterFunction.java

```diff
@@ -77,6 +77,12 @@ public static FilterFunction build(
             ? QueryNodeMapper.getInstance().getQuery(filterFunctionGrpc.getFilter(), indexState)
             : null;
     float weight = filterFunctionGrpc.getWeight() != 0.0f ? filterFunctionGrpc.getWeight() : 1.0f;
+    if (filterFunctionGrpc.hasDecayFunction()) {
+      MultiFunctionScoreQuery.DecayFunction decayFunction = filterFunctionGrpc.getDecayFunction();
+      if (decayFunction.hasGeoPoint()) {
+        return new GeoPointDecayFilterFunction(filterQuery, weight, decayFunction, indexState);
+      }
```

Comment:
If geo point is the only mode supported, an exception should probably be throw if it is not set